channeldb/kvdb/etcd/readwrite_cursor.go

```diff
@@ -0,0 +1,88 @@
+package etcd
+
+type readWriteCursor struct {
+	bucket  *readWriteBucket
+	prefix  string
+	currKey string
+}
+
+func newReadWriteCursor(bucket *readWriteBucket) *readWriteCursor {
+	return &readWriteCursor{
+		bucket: bucket,
+		prefix: makeValuePrefix(bucket.path),
+	}
+}
+
+// First positions the cursor at the first key/value pair and returns
+// the pair.
+func (c *readWriteCursor) First() (key, value []byte) {
+	if kv := c.bucket.tx.stm.First(c.prefix); kv != nil {
+		c.currKey = kv.key
+		return []byte(kv.key[len(c.prefix):]), []byte(kv.val)
```

Comment:
Similar comment here re bounds checking prefix key access before hand.

Reply:
Similarly to my above comment, `stm.First` will either return `nil` or a match which contains prefix, so I believe bounds check is unnecessary here.